test/unit/stretching-test.js

```diff
@@ -0,0 +1,98 @@
+        assert.notOk(invalid);
+
+        invalid = stretching.stretch('a', element , 1, null, 1, 1);
+        assert.notOk(invalid);
+
+        invalid = stretching.stretch('a', element , 1, 1, null, 1);
+        assert.notOk(invalid);
+
+        invalid = stretching.stretch('a', element , 1, 1, 1, null);
+        assert.notOk(invalid);
+    });
+
+    test('stretching.stretch with different stretch', function(assert) {
+        var element = document.createElement('div');
+
+        // uniform test
+        var scale = stretching.stretch(null, element , 4, 2, 1, 1);
+        assert.notOk(scale, 'uniform is not scaled');
+        assert.equal(element.className, ' jw-stretch-uniform', 'uniform stretch should have uniform class');
+
+        scale = stretching.stretch('uniform', element , 2, 4, 2, 1);
+        assert.notOk(scale, 'uniform is not scaled');
+        assert.equal(element.className, ' jw-stretch-uniform', 'uniform stretch should have uniform class');
+
+        scale = stretching.stretch('uniform', element , 2, 4, 1, 2);
+        assert.ok(scale, 'scaled with exactfit when yscale is bigger and exactfit matches');
+        assert.equal(element.className, ' jw-stretch-exactfit', 'exactfit stretch should have exactfit class');
+
+        scale = stretching.stretch('uniform', element , 4, 2, 2, 1.001);
+        assert.ok(scale, 'scaled with exactfit when xscale is bigger and exactfit matches');
+        assert.equal(element.className, ' jw-stretch-exactfit', 'exactfit stretch should have exactfit class');
+
+        // fill test
+        scale = stretching.stretch('fill', element , 2, 4, 2, 1);
+        assert.ok(scale, 'fill is always scaled');
+        assert.equal(element.className, ' jw-stretch-fill', 'fill stretch should have fill class');
+
+        scale = stretching.stretch('fill', element , 4, 2, 1, 1);
+        assert.ok(scale, 'fill is always scaled');
+        assert.equal(element.className, ' jw-stretch-fill', 'fill stretch should have fill class');
+
+        // non test
+        scale = stretching.stretch('none', element , 4, 2, 1, 1);
+        assert.ok(scale, 'none is always scaled');
+        assert.equal(element.className, ' jw-stretch-none', 'none stretch should have none class');
+    });
+
+    test('stretching.stretch with video', function(assert) {
+        var element = document.createElement('video');
+
+        // parent width/height larger
+        var scale = stretching.stretch('none', element, 4, 2, 1, 1);
+        assert.ok(scale, 'none is always scaled');
+        assert.ok(element.style.cssText.indexOf('width: 1px; height: 1px') >= 0,
+            'element should be correctly scaled');
+
+        // parent width/height smaller
+        scale = stretching.stretch('none', element, 1, 1, 4, 2);
+        assert.ok(scale, 'none is always scaled');
+        assert.ok(element.style.cssText.indexOf('left: -1px; right: -1px; width: 4px; height: 2px') >= 0,
```

Comment:
This test is failing in chrome. "element should be correctly scaled". Maybe remove the spaces or allow for less rigid matching of property values.